dhis-2/dhis-api/src/main/java/org/hisp/dhis/period/PeriodType.java

```diff
@@ -172,6 +173,19 @@ public static List<PeriodType> getAvailablePeriodTypes()
         return new ArrayList<>( PERIOD_TYPES );
     }
 
+    /**
+     * Returns an immutable list of the names of all period types in their
+     * natural order.
+     *
+     * @return an immutable list of the names of all period types.
+     */
+    public static List<String> getAvailablePeriodTypeNames()
+    {
+        return List.copyOf( PERIOD_TYPES.stream()
```

Comment:
unnecessary `copyOf`. `collect` already create a new list.

Reply:
For that there is `toUnmodifyableList()` collector.

Reply:
Right, I will update it.